cmd/fleet/main.go

```diff
@@ -245,7 +248,21 @@ func (a *AgentMode) Run(ctx context.Context) error {
 	// trigger startChan so OnConfig can continue
 	a.startChan <- struct{}{}
 
-	return a.srv.Run(srvCtx)
+	// keep trying to restart the FleetServer on failure, reporting
+	// the status back to Elastic Agent
+	res := make(chan error)
+	go func() {
+		for {
+			err := a.srv.Run(srvCtx)
+			if err == nil || err == context.Canceled {
+				res <- err
+				return
+			}
+			// sleep some before calling Run again
+			sleep.WithContext(srvCtx, kAgentModeRestartLoopDelay)
```

Comment:
if the context is cancelled on `sleep`, should we bail out and not attempt another `srv.Run`?
maybe 
```
			err = sleep.WithContext(srvCtx, kAgentModeRestartLoopDelay)
			if err != nil {
                                 res <- err
				return
			}
```

Reply:
thinking out loud: should we rethink how we start http server for the next version maybe? 
currently it's tied the fleet server, if something failed to start the whole server is down for 2 seconds.
it would be more graceful if the `/api/status` route for example would still be available reflecting the status of the fleet server if it has issues starting

Reply:
I just thought it made the code path simpler, calling `srv.Run` with a cancelled context should just make it return the error, so that is why I keep it this way. I can add the check if you want.

Yeah that probably would be nice to change that in 7.14.